[ContainerRegistry] update for better codegen for data plane Track 2 SDK libraries

jeremymeng opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify: **update swagger to allow better codegen for data plane Track 2 SDK libraries**
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

jeremymeng committed:
[Track2 Data Plane SDK] prepare folder

Add a new folder Azure.ContainerRegistry with a copy of
preview/2019-08-15 specs in preparation of further changes to make
better codegen of Track 2 data plane SDK libraries

jeremymeng committed:
Add Track 2 customizations

annelo-msft commented:
@jeremymeng , if the Track 1 swagger filepath is  `Microsoft.ContainerRegistry/preview/2019-08-15/`, would it make sense to move the Track 2 swagger to a filepath with a different date?  Or does the date mean a specific version of the service, and so the Track 2 file describes the same one and we should use the same date?

jeremymeng commented:
> Or does the date mean a specific version of the service

yes I believe it is the service version. At least that what I've been seeing in Storage.

annelo-msft committed:
update PATCH return types

jeremymeng committed:
Add a preliminary readme.md

jeremymeng committed:
Fix CI Avocado failures

- rename folder to be consistent with api version
- remove GetAccessToken.json example as the endpoint was removed in this swagger

jeremymeng committed:
Add timeasc and timedesc to custom-words.txt

also remove sections unrelated to Track 2 SDK

jeremymeng committed:
Merge remote-tracking branch 'upstream/master' into acr-track2-swagger

jeremymeng committed:
Update typescript customization

jeremymeng committed:
Fix required-ness for some properties

Fix casing aadAccesstoken => aadAccessToken

jeremymeng committed:
Run prettier

jeremymeng commented on specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md:
````diff
@@ -0,0 +1,87 @@
+These settings apply only when `--csharp` is specified on the command line.
+
+``` yaml $(csharp)
+csharp: true
````
@annelo-msft please review

jeremymeng commented on specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md:
````diff
@@ -0,0 +1,87 @@
+These settings apply only when `--java` is specified on the command line.
+
+``` yaml $(java)
+java: true
````
@pallavit could you please add your Java customization here as a suggestion comment?

jeremymeng commented on specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md:
````diff
@@ -0,0 +1,87 @@
+These settings apply only when `--python` is specified on the command line.
+
+``` yaml $(python)
+python: true
````
@seankane-msft could you please add your Python customization here as a suggestion comment?

annelo-msft committed:
updates to swagger for Beta 2 API

annelo-msft committed:
bug fix in arch enum

annelo-msft committed:
updates per apiview feedback

annelo-msft committed:
updates to manifest properties

jeremymeng committed:
Move some customization into swagger

jeremymeng committed:
Undo client name customization for `architecture`

Now that we use `architecture` in the new API design.

jeremymeng committed:
Update arch and os types and set readOnly: true

seankane-msft committed:
make change for grant type = password

seankane-msft committed:
editing the correct swagger

seankane-msft committed:
undoing changes to the wrong swagger

seankane-msft committed:
adding internal enum for grant type

seankane-msft committed:
moving grant_type to the end of the parameters for .net

jeremymeng committed:
Incorporate Anne's beta.3 swagger update

and make ManifestAttributes.manifest required so its flattened
`digest` property can have their proper required-ness.

annelo-msft committed:
small name changes

annelo-msft committed:
updates to separate out ArtifactManifestReference

annelo-msft committed:
flatten writeable property types; add registry property back

pallavit committed:
update the delete methods

jeremymeng committed:
Remove property required-ness

as they are used for input and cannot be used together with "readOnly"

jeremymeng committed:
Run prettier

jeremymeng committed:
Fix linting errors in examples

jeremymeng committed:
Remove extraneous brace

jeremymeng committed:
Add custom word for ACR

jeremymeng committed:
Fix more linting errors in examples

jeremymeng committed:
Revert "Remove property required-ness"

This reverts commit 4eba58a49bfd3a3e51219c90f583521391596218.

jeremymeng committed:
Remove default value of "none" for architecture enum

annelo-msft committed:
renames per arch board review

jeremymeng committed:
Copy the existing suppression from master

jeremymeng committed:
Fix spellcheck errors

Add words to custom-words.txt

jeremymeng commented on specification/containerregistry/data-plane/Azure.ContainerRegistry/readme.md:
````diff
@@ -0,0 +1,127 @@
+python: true
+```
+
+## Suppression
````
The Suppression section is copied from existing master branch 

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerregistry/data-plane/readme.md#suppression

jeremymeng committed:
More spelling fix

jeremymeng committed:
Suppress R2056 RequiredReadOnlyProperties

Reasoning at https://github.com/Azure/azure-openapi-validator/issues/243

jeremymeng committed:
Remove `:` in reason value

it breaks yaml

jeremymeng committed:
Update description of registryLoginServer

as `azurecr.io` is kust for public cloud and the format can be
different for every cloud or vanity URLs.

tjprescott commented:
@jeremymeng is this ready to merge?

jeremymeng commented:
> @jeremymeng is this ready to merge?

@tjprescott I contacted and would like to get some review and approval from the service team first.

shahzzzam commented on specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json:
```diff
@@ -0,0 +1,3090 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Container Registry",
```
Azure Container Registry?

jeremymeng replied:
Updated

shahzzzam commented on specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json:
```diff
@@ -0,0 +1,3090 @@
+          "readEnabled": true,
+          "listEnabled": true,
+          "deleteEnabled": true,
+          "writeEnabled": true
```
can you add `"teleportEnabled": false`?

jeremymeng replied:
Added

jeremymeng committed:
Address PR feedback

- Add "Azure" to title
- add `"teleportEnabled": false` to example

tjprescott commented:
@jeremymeng please ping me when you are ready to merge this

shahzzzam reviewed (approved):
LGTM

jeremymeng committed:
Minor update to tyepscript codegen customization section

jeremymeng commented:
@tjprescott It's ready to go!
